core/trino-main/src/main/java/io/trino/sql/gen/JoinCompiler.java

```diff
@@ -168,14 +180,22 @@ private LookupSourceSupplierFactory internalCompileLookupSourceFactory(List<Type
     {
         Class<? extends PagesHashStrategy> pagesHashStrategyClass = internalCompileHashStrategy(types, outputChannels, joinChannels, sortChannel);
 
+        OptionalInt singleBigintJoinChannel = OptionalInt.empty();
+        if (enableSingleChannelBigintLookupSource
+                && joinChannels.size() == 1
+                && types.get(getOnlyElement(joinChannels)) == BIGINT) {
+            singleBigintJoinChannel = OptionalInt.of(getOnlyElement(joinChannels));
+        }
+
         Class<? extends LookupSourceSupplier> joinHashSupplierClass = IsolatedClass.isolateClass(
                 new DynamicClassLoader(getClass().getClassLoader()),
                 LookupSourceSupplier.class,
                 JoinHashSupplier.class,
                 JoinHash.class,
+                PagesHash.class,
```

Comment:
Why this is needed? or maybe where it stats to slow down without it? This is JIT abc to optimize classes that use single implementation of interface. If this doesn't work at some level, this breaks our assumptions about Java perf

Reply:
I don't know. Just the fact that it works is enough for me